lib/spack/spack/solver/concretize.lp

```diff
@@ -192,6 +192,21 @@ attr(Name, A1, A2)         :- impose(ID), imposed_constraint(ID, Name, A1, A2).
 attr(Name, A1, A2, A3)     :- impose(ID), imposed_constraint(ID, Name, A1, A2, A3).
 attr(Name, A1, A2, A3, A4) :- impose(ID), imposed_constraint(ID, Name, A1, A2, A3, A4).
 
+% If something is mentioned as a possible constraint, allow for the possibility
+% of imposing it
+{impose(ID)} :- imposed_constraint(ID, Name, A1).
+{impose(ID)} :- imposed_constraint(ID, Name, A1, A2).
+{impose(ID)} :- imposed_constraint(ID, Name, A1, A2, A3).
+{impose(ID)} :- imposed_constraint(ID, Name, A1, A2, A3, A4).
```

Comment:
I think these rule may result in an `impose(ID)` atom where a corresponding `do_not_impose(ID)` exists. I'll try to check the PR tomorrow, but we might need:
```
:- do_not_impose(ID), impose(ID).
```
to get back the previous semantic.

Reply:
@alalazo I have added that